### PR TITLE
Remove documentation of legacy `frontend_proxy` worker app

### LINKED ISSUE #3717: frontend_proxy worker claims it forwards requests to the main process, but it doesn't
Having ensured I set up the `worker_main_http_uri` to point to the main process, making a request to something like `/_matrix/client/versions` results in a 400 unrecognized request. There's no evidence that it actually tries to contact the main process (no additional logging in the worker or main process)

### PR DESCRIPTION
**Pull request by richvdh**
This has been the same as a generic_worker since #6964, so let's get rid of it.

Fixes #3717

Supercedes #13451.

### CODE REVIEW

File: docs/workers.md [hunk 1]
```diff
@@ -581,39 +583,16 @@ handle it, and are online.
...
-```
-
 ### Historical apps
 
-*Note:* Historically there used to be more apps, however they have been
-amalgamated into a single `synapse.app.generic_worker` app. The remaining apps
-are ones that do specific processing unrelated to requests, e.g. the `pusher`
-that handles sending out push notifications for new events. The intention is for
-all these to be folded into the `generic_worker` app and to use config to define
-which processes handle the various proccessing such as push notifications.
```
Comment (richvdh) [inline docs/workers.md:616]: I don't think this information about intended direction is terribly helpful here: users aren't interested in the five-year-plan; they want to know how it works today.

File: docs/workers.md [hunk 2]
```diff
@@ -117,23 +117,23 @@ redis:
...
-`worker_listeners` option should be set with a `http` listener, in the same way
-as the `listeners` option in the shared config.
+In the config file for each worker, you must specify:
+ * The type of worker (`worker_app`). The currently available worker applications are listed below.
+ * A unique name for the worker (`worker_name`).
+ * The HTTP replication endpoint that it should talk to on the main synapse process
+   (`worker_replication_host` and `worker_replication_http_port`)
+ * If handling HTTP requests, a `worker_listeners` option with an `http`
+   listener, in the same way as the `listeners` option in the shared config.
+ * If handling the `^/_matrix/client/v3/keys/upload` endpoint, the HTTP URI for
```
Comment (H-Shay) [inline docs/workers.md:135]: For my own edification-why is this required? (Not questioning, just curious)

**Review by H-Shay — APPROVED** [review]
lgtm

File: docs/workers.md (hunk 2 above)
Comment (richvdh) [reply docs/workers.md:135]: which bit, in particular?

Comment (richvdh) [reply docs/workers.md:135]: @H-Shay I guess you're asking about `worker_main_http_uri`? In which case, the answer is that `/_matrix/client/v3/keys/upload` is sometimes processed locally on the worker, but sometimes needs passing over to the main process; and the way it does the latter is very hacky and just an HTTP hit on the main process's regular HTTP interface.

https://github.com/matrix-org/synapse/issues/13642 is about improving this.

Comment (H-Shay) [reply docs/workers.md:135]: That was it, thank you for explaining!
